samples/bluetooth/peripheral_gap_svc/src/main.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2025 Koppel Electronic
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/gatt.h>
+
+
+#define DEVICE_NAME		CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN		(sizeof(DEVICE_NAME) - 1)
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
```

Comment:
If `CONFIG_BT_DEVICE_NAME_GATT_WRITABLE=y` then we shouldn't use `CONFIG_BT_DEVICE_NAME` for the advertising data, as if it's changed then we are advertising an incorrect name